src/Keys/Version4/AsymmetricSecretKey.php

```diff
@@ -94,11 +97,25 @@ public function getPublicKey(): AsymmetricPublicKey
      *
      * @throws Exception
      */
-    public static function importPem(string $pem, ProtocolInterface $protocol = null): self
+    public static function importPem(string $pem, ?ProtocolInterface $protocol = null): self
     {
         $formattedKey = str_replace('-----BEGIN EC PRIVATE KEY-----', '', $pem);
         $formattedKey = str_replace('-----END EC PRIVATE KEY-----', '', $formattedKey);
-        $key = Base64::decode(strtok($formattedKey, "\n"));
+
+        /**
+         * @psalm-suppress DocblockTypeContradiction
+         * PHP 8.4 updated the docblock return for str_replace, which makes this check required
+         */
+        if (!is_string($formattedKey)) {
+            throw new PasetoException('Invalid PEM format', ExceptionCode::UNSPECIFIED_CRYPTOGRAPHIC_ERROR);
+        }
+
+        $tokenizedKey = strtok($formattedKey, "\n");
```

Comment:
Is this safe to do?

https://github.com/php/php-src/blob/06c41ec6c54a2b6d79d91976041ef12241550683/main/reentrancy.c#L263-L322

https://github.com/walac/glibc/blob/264aad1e6aca30efddfcfae05c44ad38bb5e137d/string/strtok.c#L25-L65

https://github.com/walac/glibc/blob/264aad1e6aca30efddfcfae05c44ad38bb5e137d/string/strpbrk.c#L28-L44

It sure seems like this would expose a timing leak that exposes the private key.

Reply:
Is the issue the creation of the variable to store the value from `strtok`?  The `decode` function requires a `string`, whereas `strtok` returns `false` on error.  So it's necessary to check for that and not attempt to pass `false` to the `decode` function.

That said, if the issue is the variable, we could do something like...

```php
$key = Base64::decode(
    strtok($formattedKey, "\n") ?: throw new PasetoException(
        'Invalid PEM format', 
        ExceptionCode::UNSPECIFIED_CRYPTOGRAPHIC_ERROR,
    ),
);
```

Reply:
No, the problem is `strtok()` itself.

Reply:
Okay, that's something I'm unsure of, as my C skills are a bit lacking.  But seeing as that's not a new concern, I think this PR could be merged and `strtok` can be revisited in a new issue ticket/PR.